### ISSUE TITLE
`consider-using-f-string` should not be triggered by format strings including length

### ISSUE THREAD

**Issue by dgutson** [pylint-dev/pylint#8864]
### Bug description

# pylint: disable=consider-using-f-string

I think pylint should not trigger this warning with strings like

`"%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x"`

since using an f-string would be much more complicated.

### Command used

```shell
pylint default
```

### Pylint output

```shell
pylint_bug.py:2:7: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
```

### Expected behavior

none

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
```

**Comment by mbyrnepr2** [comment]
Thanks for the issue. Checking if we are on the same page regarding a comparison to the example of the f-string version; is this what you had in mind?
```python
print("%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x" % (1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
print(f"{1:08x}-{2:04x}-{3:04x}-{4:04x}-{5:02x}{6:02x}{7:02x}{8:02x}{9:02x}{10:02x}")
```

**Comment by dgutson** [comment]
Roughly, imagine that the values come in variables. I arguably see cleaner the first form since things are separsted.

But regardless of the preferred style, which ultimately depends on the developer coding guideline he/she adheres to, I think that pylint should allow that form. Or at least have a flag allowing it.

**Comment by mbyrnepr2** [comment]
It looks like a duplicate of https://github.com/pylint-dev/pylint/issues/8296. What do you think @dgutson? If they are the same in principle, it’s easier to manage things if we have all the info in one place. We could copy your example to that issue.

**Comment by dgutson** [comment]
> It looks like a duplicate of #8296. What do you think @dgutson? If they are the same in principle, it’s easier to manage things if we have all the info in one place. We could copy your example to that issue.

Ok, it's a bit different use case but the same solution would work for me.
I agree about marking this issue as duplicate and copy the example.
Sorry that I didn't find that one on my search when I checked before filing this issue.
